### PR TITLE
Configure CrepLogger

### LINKED ISSUE #30: Use logger instead of `puts`
Currently we use `puts` in Crep and the [XINGLogger](https://github.com/xing/crep/blob/master/lib/crep.rb#L7) is not used at all.

Replace all usages of puts with the defined XINGLogger.

This would allow us to better handle `verbose` output.

### PR DESCRIPTION
**Pull request by dhartwich1991**
## Changes: 
- Rename XINGLogger to `CrepLogger`
- Configuration for the Logger (colorization)
- Replaced `puts` in some places
- First implementation for #44 
- Closes #30 

Screenshot of an example log: 
daniel_hartwich_jenkins-slave-2____android-app

### CODE REVIEW

File: lib/crep.rb [hunk 1]
```diff
@@ -4,8 +4,26 @@
 require 'logger'
 
 module Crep
-  XINGLogger = Logger.new(STDOUT)
-  XINGLogger.level = Logger::DEBUG
+  CrepLogger = Logger.new(STDOUT)
```
Comment (BasThomas) [inline lib/crep.rb]: Would we want to extract this logic to its own file?

File: lib/crep.rb [hunk 2]
```diff
@@ -4,8 +4,26 @@
...
+  CrepLogger.level = Logger::DEBUG
+  CrepLogger.datetime_format = '%Y-%m-%d %H:%M:%S'
+  CrepLogger.formatter = proc do |severity, datetime, progname, msg|
+    case severity
+    when "DEBUG"
+      "\e[35m[DEBUG] #{datetime}: #{msg}\033[0m\n"
```
Comment (BasThomas) [inline lib/crep.rb]: Whats the `\e[35m` and `\033[0m` doing?

File: lib/crep.rb [hunk 3]
```diff
@@ -4,8 +4,26 @@
...
+    when "ERROR"
+      "[ERROR] #{datetime}: #{msg}\n".red
+    when "FATAL"
+      "[FATAL] #{datetime}: #{msg}\n".red
+    when "INFO"
+      "[INFO] #{datetime}: #{msg}\n".green
+    when "WARN"
+      "[WARNING] #{datetime}: #{msg} \n".yellow
```
Comment (BasThomas) [inline lib/crep.rb]: Should remove the trailing space.

File: lib/crep.rb (hunk 2 above)
Comment (dhartwich1991) [reply lib/crep.rb]: It's a manual way of color formatting (this one is purple-ish)
It follows the Ansi Escape Code Table. So: `\e[35m` sets the color to purple for all the following input/output and `\0ee[0m` resets it to the default.
ansi_escape_code_-_wikipedia

Comment (dhartwich1991) [reply lib/crep.rb]: Arghh should have checked the `colorized` gem before
``` ruby
    def color_codes
      {
        :black   => 0, :light_black    => 60,
        :red     => 1, :light_red      => 61,
        :green   => 2, :light_green    => 62,
        :yellow  => 3, :light_yellow   => 63,
        :blue    => 4, :light_blue     => 64,
        :magenta => 5, :light_magenta  => 65,
        :cyan    => 6, :light_cyan     => 66,
        :white   => 7, :light_white    => 67,
        :default => 9
```

Using `light_magenta` now instead.

**Pull request merged**